magniv/build/build.py

```diff
@@ -32,44 +30,46 @@ def build():
-                mod = importlib.util.module_from_spec(spec)
-                spec.loader.exec_module(mod)
-                tasks = getmembers(mod, is_task)
-                req = (
-                    "{}/requirements.txt".format(root)
-                    if os.path.exists("{}/requirements.txt".format(root))
-                    else root_req
-                )
-                if req == None:
-                    raise OSError(
-                        f'requirements.txt not found for path "{root}", either add one to this directory or the root directory'
-                    )
-                for name, t in tasks:
-                    if t.key in used_keys:
-                        raise ValueError(
-                            f'Task "{t.key}" in file {filepath} is using "{t.key}" as a key which is already used in {used_keys[t.key]}, please resolve by changing one of the keys'
-                        )
-                    else:
-                        used_keys[t.key] = filepath
-                    code, lineno = getsourcelines(t)
-                    info = {
-                        "location": filepath,
-                        "name": t.name,
-                        "key": t.key,
-                        "description": t.description,
-                        "schedule": t.schedule,
-                        "python_version": _get_python_version(root),
-                        "owner": _get_owner(root),
-                        "requirements_location": req,
-                        "line_number": lineno,
-                    }
-                    tasks_list.append(info)
+        if ext == ".py":
+            filepath = "{}/{}".format(root, file_name)
+            with open(filepath) as file:
+                parsed_ast = ast.parse(file.read())
+                for node in ast.walk(parsed_ast):
+                    if isinstance(node, (ast.FunctionDef, ast.AsyncFunctionDef)):
+                        if len(node.decorator_list) > 0:
+                            for decorator in node.decorator_list:
+                                if (
+                                    not isinstance(decorator, ast.Name)
+                                    and decorator.func.id == "task"
+                                ):
+                                    for decorator in node.decorator_list:
+                                        info = {
+                                            "location": filepath,
+                                            "name": node.name,
+                                            "python_version": _get_python_version(root),
+                                            "owner": _get_owner(root),
+                                            "requirements_location": req,
+                                            "line_number": node.lineno,
+                                        }
+                                        for kw in decorator.keywords:
```

Comment:
This may be problematic, b/c in another parts of the code where we ingest this information we expect `description`, `key`, and `schedule` to exist.

`schedule` is for sure needed and there needs to be an error thrown if it doesn't exist.

Reply:
ah kk check my latest commit

Reply:
The only one that should be in `missing_reqs` is `schedule`,  if `key` doesnt exist we default it to the name of the function, you can add `key` to the `info` object on line 57 and have it default to the name

Reply:
fixed

Reply:
This looks good, is it ready to merge?